diff --git a/node_modules/@walletconnect/jsonrpc-utils/dist/esm/format.js b/node_modules/@walletconnect/jsonrpc-utils/dist/esm/format.js
index 325f82e..3aab73b 100644
--- a/node_modules/@walletconnect/jsonrpc-utils/dist/esm/format.js
+++ b/node_modules/@walletconnect/jsonrpc-utils/dist/esm/format.js
@@ -1,9 +1,12 @@
 import { getError, getErrorByCode, isReservedErrorCode } from "./error";
 import { INTERNAL_ERROR, SERVER_ERROR } from "./constants";
+function getRandomInt(min, max) {
+    min = Math.ceil(min);
+    max = Math.floor(max);
+    return Math.floor(Math.random() * (max - min + 1)) + min;
+}
 export function payloadId(entropy = 3) {
-    const date = Date.now() * Math.pow(10, entropy);
-    const extra = Math.floor(Math.random() * Math.pow(10, entropy));
-    return date + extra;
+    return getRandomInt(0, Number.MAX_SAFE_INTEGER);
 }
 export function getBigIntRpcId(entropy = 6) {
     return BigInt(payloadId(entropy));
